pkg/scaling/resolver/azure_keyvault_handler.go

```diff
@@ -125,12 +128,12 @@ func (vh *AzureKeyVaultHandler) getAuthConfig(ctx context.Context, client client
 	case kedav1alpha1.PodIdentityProviderAzure:
 		config := auth.NewMSIConfig()
 		config.Resource = keyVaultResourceURL
-		config.ClientID = vh.podIdentity.IdentityID
+		config.ClientID = vh.vault.Credentials.PodIdentity.IdentityID
```

Comment:
```suggestion
		config.ClientID = podIdentity.IdentityID
```

Reply:
Good catch, I added the validation testing it at the end, and I forgot to change other places